version: '3.8'


services:
    dbase-service:
        build:
            context: ./dbase-service
            dockerfile: Dockerfile
        image: dbase-service
        container_name: dbase-api-service
        environment:
            - POSTGRES_HOST=dbase-api-service
            - POSTGRES_PORT=5678
            - POSTGRES_USER=roostique
            - POSTGRES_PASSWORD=password
            - POSTGRES_DB=postgres
        command: -p 3333
        networks:
            custom-network:
                ipv4_address: 172.20.0.7
        healthcheck:
            test: [ "CMD-SHELL",
                   "pg_isready
                    -h dbase-api-service
                    -p 3333
                    -d postgres
                    -U roostique"
            ]
            interval: 10s
            timeout: 5s
            retries: 5

    api-service:
        build:
            context: ./api-service/service
            dockerfile: Dockerfile
        image: api-service
        container_name: api-service
        environment:
            - POSTGRES_HOST=dbase-service
            - POSTGRES_PORT=3333
            - POSTGRES_USER=postgres
            - POSTGRES_PASSWORD=password
            - POSTGRES_DB=postgres
            - APP_HOST=api-service
            - APP_PORT=1477
        networks:
            custom-network:
                ipv4_address: 172.20.0.8
        healthcheck:
            test: [
                "CMD-SHELL",
                "curl --fail http://api-service:1477/ping || exit 1"
            ]
            interval: 10s
            timeout: 5s
            retries: 3
        depends_on:
            dbase-service:
                condition: service_healthy

networks:
    custom-network:
       driver: bridge
       ipam:
          driver: default
          config:
              - subnet: 172.20.0.0/24
                gateway: 172.20.0.1
